packages/headless/src/controllers/insight/search-box/headless-insight-search-box.ts

```diff
@@ -40,9 +48,41 @@ export function buildSearchBox(
   engine: InsightEngine,
   props: SearchBoxProps = {}
 ): SearchBox {
-  return buildCoreSearchBox(engine, {
+  const {dispatch} = engine;
+  const searchBox = buildCoreSearchBox(engine, {
     ...props,
     executeSearchActionCreator: executeSearch,
     fetchQuerySuggestionsActionCreator: fetchQuerySuggestions,
   });
+  const id = props.options?.id || randomID('search_box');
+  const options: Required<SearchBoxOptions> = {
+    id,
+    highlightOptions: {...props.options?.highlightOptions},
+    ...defaultSearchBoxOptions,
+    ...props.options,
+  };
+
+  const getValue = () => engine.state.querySet?.[options.id];
+
+  const performSearch = async (analytics: SearchAction) => {
+    const {enableQuerySyntax, clearFilters} = options;
+
+    dispatch(
+      prepareForSearchWithQuery({
+        q: getValue(),
+        enableQuerySyntax,
+        clearFilters,
+      })
+    );
+
+    await dispatch(executeSearch(analytics));
```

Comment:
It would be nice to add a test to ensure the analytics action is the one being dispatched.

Reply:
Sure thing 👌

Reply:
tests added 👌